index.html

```diff
@@ -4072,63 +4072,104 @@ <h2>Elements</h2>
 
-  <li>If <var>node id map</var> does not contain <var>node</var>,
-  set <var>node id map</var>[<var>node</var>] to a new globally
-  unique string.
+  <li>If <var>node id map</var> does not contain <var>node</var>:
+
+    <ol>
+      <li>Let <var>node id</var> be a new globally unique string.
+
+      <li>Set <var>node id map</var>[<var>node</var>] to <var>node id</var>.
+
+      <li>Let <var>navigable</var> be <var>browsing
+      context</var>'s <a>active document</a>'s <a>node navigable</a>.
+
+      <li>Let <var>navigable seen nodes map</var>
+      be <var>session</var>'s <a>navigable seen nodes map</a>.
 
+      <li>If <var>navigable seen nodes map</var> does not
+      contain <var>navigable</var>, set <var>navigable seen nodes
+      map</var>[<var>navigable</var>] to an empty set.
+
+      <li>Append <var>node id</var> to <var>navigable seen nodes
+      map</var>[<var>navigable</var>].
+    </ol>
  <li><p>Return <var>node id map</var>[<var>node</var>].
 </ol>
 
+<p>A <dfn>node reference is known</dfn> given <var>session</var>, <var>browsing
+context</var>, and <var>reference</var> if the following steps return true:
+<ol class="algorithm">
+  <li>Let <var>navigable</var> be <var>browsing
+  context</var>'s <a>active document</a>'s <a>node navigable</a>.
+
+  <li>Let <var>navigable seen nodes map</var>
+  be <var>session</var>'s <a>navigable seen nodes map</a>.
+
+  <li>If <var>navigable seen nodes map</var>
+  [=map/contains=] <var>navigable</var> and <var>navigable seen nodes
+  map</var>[<var>navigable</var>] [=set/contains=] <var>reference</var>,
+  return true, otherwise return false.
+</ol>
+
 <p>To <dfn>get a known element</dfn> given <var>reference</var>:
-<ol>
-  <li>Let <var>node</var> be <a>get a node</a> with <a>current session</a>
-  and <var>reference</var>.
+<ol class="algorithm">
+  <li>Let <var>node</var> be <a>get a node</a> with <a>current session</a>,
+  <a>current browsing context</a>, and <var>reference</var>.
+
+  <li>If <var>node</var> is null, or <var>node</var> does not
+  implement {{Element}} return <a>error</a> with <a>error
+  code</a> <a>no such element</a>.
 
-  <li>If <a>node</a> is null, or <a>node</a> does not implement {{Element}},
-  or if <a>node</a>'s [=Node/node document=] is not <a>current browsing
-  context</a>'s <a>active document</a>, return <a>error</a>
-  with <a>error code</a> <a>no such element</a>.
+  <li>If <var>node</var> <a>is stale</a>:
+   <ol>
+     <li><p>If <a>node reference is known</a> with <a>current
+     session</a>, <a>current browsing context</a>,
+     and <var>reference</var>, return <a>error</a> with <a>error
+     code</a> <a>stale element reference</a>.
 
-  <li>If <var>node</var> <a>is stale</a>, return <a>error</a>
-  with <a>error code</a> <a>stale element reference</a>.
+     <p>Otherwise return <a>error</a> with <a>error
+     code</a> <a>no such element</a>.
+   </ol>
 
   <li>Return <a>success</a> with <var>node</var>.
```

Comment:
As  you said on another PR shouldn't this be `Return success with data`? If that's the case we also need that phrasing for `get a known shadow root`.